src/librustc/session/mod.rs

```diff
@@ -292,6 +292,30 @@ impl Session {
             &self.opts.search_paths,
             kind)
     }
+
+    #[cfg(not(target_os = "nacl"))]
+    pub fn expect_cross_path(&self) -> PathBuf {
+        use std::env;
+        let cross_path = self.opts.cg.cross_path.clone();
+        match cross_path.or_else(|| env::var("NACL_SDK_ROOT").ok() ) {
+            None => self.fatal("need cross path (-C cross-path, or via NACL_SDK_ROOT) \
+                                for this target"),
+            Some(p) => Path::new(&p).to_path_buf(),
+        }
+    }
+    #[cfg(not(target_os = "nacl"))]
+    pub fn pnacl_toolchain(&self) -> PathBuf {
+        let mut tc = self.expect_cross_path();
+        tc.push("toolchain");
+        tc.push(&format!("{}_pnacl", get_os_for_nacl_toolchain(self)));
+        tc
+    }
+
+    /// Shortcut to test if we need to do special things because we are targeting PNaCl.
+    pub fn targeting_pnacl(&self) -> bool {
+        &self.target.target.target_os[..] == "nacl"
+            && &self.target.target.arch[..] == "le32"
+    }
```

Comment:
I'd rather not have pnacl-specific stuff in the session. Is there somewhere more pnacl specific in the compiler we can bury these. Maybe in the backend where the other platform-specific things live like target specs.